ballerina/tests/client_oauth2_provider_test.bal

```diff
@@ -875,3 +875,57 @@ isolated function testAccessTokenRequestWithSecureSocketAndWithHttpUrlScheme() r
     string response = check provider.generateToken();
     assertToken(response);
 }
+
+@test:Config {}
+isolated function testInvalidTokenUrl() returns Error? {
+    ClientCredentialsGrantConfig config = {
+        tokenUrl: "",
+        clientId: "FlfJYKBD2c925h4lkycqNZlC2l4a",
+        clientSecret: "PJz0UhTJMrHOo68QQNpvnqAY_3Aa",
+        scopes: ["view-order"],
+        clientConfig: {
+            secureSocket: {
+                cert: {
+                    path: TRUSTSTORE_PATH,
+                    password: "ballerina"
+                }
+            }
+        }
+    };
+    ClientOAuth2Provider|error provider = trap new (config);
```

Comment:
The better approach would be to return ClientOAuth2Provider| error from the ClientOAuth2Provider init() function similar to the Listener's and Client's init() functions. But it will introduce a breaking change at this point. However let's track that improvement in another issue and attend it next ballerina major release if breaking changes are possible

Reply:
Issue created: https://github.com/ballerina-platform/ballerina-standard-library/issues/3494